src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1440,7 +1454,7 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
 
                 foreach (var package in packageIdentities)
                 {
-                    var packageDependencyInfo = await dependencyInfoResource.ResolvePackage(package, nuGetFramework, CancellationToken.None);
+                    var packageDependencyInfo = await dependencyInfoResource.ResolvePackage(package, nuGetFramework, Logging.NullLogger.Instance, CancellationToken.None);
```

Comment:
similar issue with using null logger, I would make the method take a logger, and have the call site decide what to do. We probably need a bug tracking logging into output if things go wrong, there is no way to analyze the situation